autoPyTorch/pipeline/components/training/trainer/RowCutOutTrainer.py

```diff
@@ -40,22 +40,23 @@ def data_preparation(self, X: np.ndarray, y: np.ndarray,
         indices = self.random_state.choice(range(1, size), max(1, np.int32(size * self.patch_ratio)),
                                            replace=False)
 
-        if not isinstance(self.numerical_columns, typing.Iterable):
+        """if not isinstance(self.numerical_columns, typing.Iterable):
             raise ValueError("{} requires numerical columns information of {}"
                              "to prepare data got {}.".format(self.__class__.__name__,
                                                               typing.Iterable,
                                                               self.numerical_columns))
         numerical_indices = torch.tensor(self.numerical_columns)
```

Comment:
```suggestion
        numerical_indices = torch.tensor([index for index in indices if index in self.numerical_columns)
```

We can use this code and then the bug should be resolved